src/clj_irods/cache_tools.clj

```diff
@@ -82,11 +79,13 @@
   in the specified `pool`. `action` should return something that can be
   deref'd, and will be."
   [cache action pool ks]
-  (log/info "cache or agent:" ks)
   (let [cached (get-in @cache ks)]
-    (if (and (delay? cached) (realized? cached))
-      (delay (rethrow-if-error @cached))
+    (if (delay? cached) ;; here, only check that something's there, deref later is fine
```

Comment:
I realized that we should let it return a delay that's created but not realized, because if it doesn't, it just... creates the same one and then fails to put it into the cache because there's something already there.